tensorflow/compiler/xla/service/hlo_parser.cc

```diff
@@ -304,7 +304,11 @@ class HloParserImpl : public HloParser {
   bool ParseSingleInstruction(HloModule* module);
 
   // Parses a module, returning false if an error occurred.
-  bool ParseHloModule(HloModule* module);
+  // if `parse_computation_without_header` is true, the parsed text is sequence of 
+  // computations, and assume computation with `ENTRY` annotation or the last computation
+  // as module's entry computation, also using the entry computation's parameter and `ROOT`
+  // instruction's layout as module's layout. 
+  bool ParseHloModule(HloModule* module, bool parse_computation_without_header = false);
```

Comment:
Maybe add a test which starts with ENTRY as well? And the one with multiple computations?

Reply:
@cheshire I've add the tests you suggested. Could you take a look? 
Thanks.